test/databases/transaction_database.cpp

```diff
@@ -544,293 +544,317 @@ BOOST_AUTO_TEST_CASE(transaction_database_with_cache__confirm2__single_transacti
-////    };
-////
-////    // tx3: spends coinbase, tx1/1 as input, dummy output
-////    chain::input::list tx3_inputs
-////    {
-////        { { hash1, 1 }, {}, 0 }
-////    };
-////
-////    chain::output::list tx3_outputs
-////    {
-////        { 1200, {} }
-////    };
-////
-////    chain::transaction tx2(version, locktime, tx2_inputs, tx2_outputs);
-////    const auto hash2 = tx2.hash();
-////    instance.store(tx2, 1);
-////
-////    chain::transaction tx3(version, locktime, tx3_inputs, tx3_outputs);
-////    const auto hash3 = tx3.hash();
-////    instance.store(tx3, 1);
-////
-////    BOOST_REQUIRE(!instance.get(hash2).transaction().outputs().front().metadata.candidate_spend);
-////
-////    BOOST_REQUIRE_EQUAL(instance.get(hash1).height(), machine::rule_fork::unverified);
-////    BOOST_REQUIRE(!instance.get(hash1).transaction().outputs().front().metadata.spent(123, false));
-////
-////    // Setup end
-////
-////    const bool confirm1 = instance.confirm(instance.get(hash1).link(), 23, 56, 89);
-////    BOOST_REQUIRE(confirm1);
-////    const bool confirm2_3 = instance.confirm({tx2, tx3}, 123, 456);
-////    BOOST_REQUIRE(confirm2_3);
-////
-////    const auto tx2_reloaded = instance.get(hash2);
-////
-////    BOOST_REQUIRE_EQUAL(tx2_reloaded.height(), 123);
-////    BOOST_REQUIRE_EQUAL(tx2_reloaded.median_time_past(), 456);
-////    BOOST_REQUIRE_EQUAL(tx2_reloaded.position(), 0);
-////    BOOST_REQUIRE(!tx2_reloaded.candidate());
-////    BOOST_REQUIRE(!tx2_reloaded.transaction().outputs().front().metadata.candidate_spend);
-////
-////    const auto tx3_reloaded = instance.get(hash3);
-////
-////    BOOST_REQUIRE_EQUAL(tx3_reloaded.height(), 123);
-////    BOOST_REQUIRE_EQUAL(tx3_reloaded.median_time_past(), 456);
-////    BOOST_REQUIRE_EQUAL(tx3_reloaded.position(), 1);
-////    BOOST_REQUIRE(!tx3_reloaded.candidate());
-////    BOOST_REQUIRE(!tx3_reloaded.transaction().outputs().front().metadata.candidate_spend);
-////
-////    const auto tx1_reloaded = instance.get(hash1);
-////
-////    BOOST_REQUIRE_EQUAL(tx1_reloaded.height(), 23);
-////    BOOST_REQUIRE_EQUAL(tx1_reloaded.median_time_past(), 56);
-////    BOOST_REQUIRE_EQUAL(tx1_reloaded.position(), 89);
-////    BOOST_REQUIRE(!tx1_reloaded.candidate());
-////
-////    BOOST_REQUIRE_EQUAL(tx1_reloaded.transaction().outputs().front().metadata.confirmed_spend_height, 123);
-////    BOOST_REQUIRE(tx1_reloaded.transaction().outputs().front().metadata.spent(123, false));
-////}
+BOOST_AUTO_TEST_CASE(transaction_database_with_cache__confirm1__block_transactions_with_inputs_in_db__success)
+{
+   uint32_t version = 2345u;
+   uint32_t locktime = 0xffffffff;
+
+   test::create(file_path);
+   transaction_database instance(file_path, 1000, 50, 100);
+   BOOST_REQUIRE(instance.create());
+
+   // tx1: coinbase transaction
+   const chain::input::list tx1_inputs
+   {
+       { chain::point{ null_hash, chain::point::null_index }, {}, 0 }
+   };
+
+   chain::output::list tx1_outputs
```

Comment:
Use `const` (and blow), or make `tx1_inputs` non-const also and use the tx move construction.